src/core/deviceio/cpp/frame_metadata_tracker_oak.cpp

```
@@ -38,43 +45,79 @@ class FrameMetadataTrackerOak::Impl : public ITrackerImpl
                 {
-                    fb->UnPackTo(&stream.data);
-                    stream.last_timestamp = sample.timestamp;
+                    continue;
                 }
+
+                FrameMetadataOakT parsed;
+                fb->UnPackTo(&parsed);
+                stream.pending_records.push_back({ std::move(parsed), sample.timestamp });
+            }
+
+            if (!stream.pending_records.empty())
```

Cppcheck (2.19.0)

[style] The function 'get_stream_count' is never used.

(unusedFunction)